growthbook/growthbook
# [SDK][Back-end] URL Redirect Testing

### Features and Changes

Adds a way to run URL redirect tests on GrowthBook. URL redirect tests are a new type of linked change to an experiment that use Visual Changes under the hood. SDK users can customize the navigate function depending on where they are using the SDK (i.e. a user can customize the navigate function to use the router in a SPA) 

### Dependencies

#2121 Redirect Testing UI

### Testing

1. Setup a basic Create React App
2. Implement the Growthbook SDK on the app
3. Build the JS SDK locally
4. Use yarn link to reference the local build from the Growthbook backend and your sample app
5. Create a URL redirect test and make sure you are redirected as expected
6. Once you confirm basic redirects work, play around with enabling `antiFlicker` on the SDK and making sure query strings are persisted as expected when requested.
